qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -289,27 +305,35 @@ def observable(me_table, init_term=0, mapping="jordan_wigner", wires=None):
 
     sp_op_shape = (3,)
     tp_op_shape = (5,)
-    for i_table in me_table:
-        if np.array(i_table).shape not in (sp_op_shape, tp_op_shape):
-            raise ValueError(
-                "expected entries of 'me_table' to be of shape (3,) or (5,) ; got {}".format(
-                    np.array(i_table).shape
-                )
-            )
 
     # Initialize the FermionOperator
     mb_obs = FermionOperator() + FermionOperator("") * init_term
 
-    for i in me_table:
+    for table in matrix_elements:
 
-        if i.shape == (5,):
-            # two-particle operator
-            mb_obs += FermionOperator(
-                ((int(i[0]), 1), (int(i[1]), 1), (int(i[2]), 0), (int(i[3]), 0)), i[4]
+        if len(table.shape) != 2:
+            raise ValueError(
+                "Expected dimension for arrays in 'matrix_elements' is 2; got {}".format(
+                    table.shape
+                )
             )
-        elif i.shape == (3,):
-            # single-particle operator
-            mb_obs += FermionOperator(((int(i[0]), 1), (int(i[1]), 0)), i[2])
+
+        for i in table:
+
+            if np.array(i).shape not in (sp_op_shape, tp_op_shape):
+                raise ValueError(
+                    "Expected entries of matrix element tables to be of shape (3,) or (5,); got {}".format(
+                        np.array(i).shape
+                    )
+                )
+
+            if i.shape == (5,):
```

Comment:
Minor thing, but I believe the checking could be done outside of the inner loop

Reply:
Yes!. Thanks @trbromley. Indeed , this removes the overhead of looping over the conditional statements!.